default-plugins/status-bar/src/mode_info.rs

```diff
@@ -0,0 +1,187 @@
+use super::actions::{Action, Direction};
+use super::data::{InputMode, Key};
+use std::collections::HashMap;
+
+// const fn now does not support PartialEq/Eq, we have to implement our own compare fn
+const fn compare_key(l: &Key, r: &Key) -> bool {
```

Comment:
These are really cool (and I also realize I told you to try using these when this code was still in Zellij :sweat_smile: ), but if you'd like, the `State` struct here is secretly stored in a `lazy_static!` already! You could go back to your shorter first solution and pop the HashMap in there and implement either `Default` or populate it during the call to `load()`.

Sorry again to do the 180 on you here, but that would probably clean things back up a little!

Reply:
I tried to put `key_order` in State. However, it's necessary to pass `Hashmap` everywhere and it's really annoying. I prefer a `lazy_static` in `mode_info.rs` or just use current const.

Reply:
I'll give this a more thorough review today, but I think this vastly simplifies these two `const fn`s into one! (Let me know if this actually is not doing the right thing :sweat_smile:)

```rust
const fn get_key_order(key: &Key) -> Option<i32> {
    match key {
        Key::Left | Key::Right => Some(0),
        Key::Up | Key::Down => Some(1),
        Key::PageUp | Key::PageDown => Some(2),
        Key::Char(_) => Some(3),
        _ => None,
    }
}
```

Match doesn't actually need `Eq` to function in this basic manner :)

Reply:
you're right. this looks awesome!